### ISSUE TITLE
Add tag filter option

### ISSUE THREAD

**Issue by sksamuel** [kotest/kotest-gradle-plugin#4]

**Comment by rezammalik** [comment]
@sksamuel is this something that can be added? I tried looking at the logic to see if I can try adding it but got nowhere.

**Comment by rezammalik** [comment]
Sam just curious on the status on this one? I can run the tests filtering via the `gradle test --tests <filter>` but kotest doesn't honor it. Is there a workaround till this gets in to filter via kotest or this is it?

**Comment by sksamuel** [comment]
The latest snapshot has support for tags via either command line eg `./gradle kotest --tags foo` or by setting tags inside a gradle build file configuration block, eg

```
kotest {
   tags.set("LINUX && POSTGRES")
}
```

I am unsure if this works on gradle 6.x but it works on gradle 7.x

If you want to test the snapshot and I could do a 0.3.8 release.

**Comment by sksamuel** [comment]
I just remembered we don't have snapshots for gradle plugins. So I've released 0.3.8 with tags.
Feedback still very much useful.

**Comment by rezammalik** [comment]
It looks like its filtering the non-tagged and non-matching tests out correctly but it is also listing the class paths as headings for the specs whose tests it is ignoring as well. So something like this which has a test each in 2 different specs and the first one has the tag I want and the second one does not. I was doing a spec level tag for the whole spec.

```
2.  testy.TestClass1
    + Account created via endpoint
        + for user2
            + on Internal Endpoint
                - returns the correct response and passes data validation OK (1411ms)
            + on Gateway Endpoint
                - returns the correct response and passes data validation OK (1577ms)

3.  testy.TestClass2     <---- this is the line that I think should not be reported?

Specs:   2 passed, 0 failed, 3 total.  <--- maybe the total can be 1 spec that matched?
Tests:   0 passed, 0 failed, 0 ignored, 0 total.  <--- the total and passed should be 1 since 1 test per spec file and 1 spec ignored
Time:    27s`

```

**Comment by sksamuel** [comment]
Yeah Kotest should not report specs which had zero eligible tests.

Lets create another ticket for that.

**Comment by sksamuel** [comment]
https://github.com/kotest/kotest-gradle-plugin/issues/16

**Comment by rezammalik** [comment]
The total passed also is kinda suspect there. Do we need a separate one for that?

**Comment by sksamuel** [comment]
Will be fixed automatically on the other fix.

**Comment by rezammalik** [comment]
I forgot to mention this works on gradle 7 and did not on 6.8. The specific error was 

```
> Task :kotest FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':kotest'.
> org.gradle.process.internal.DefaultExecActionFactory.of(Lorg/gradle/api/internal/file/FileResolver;Lorg/gradle/api/internal/file/FileCollectionFactory;Lorg/gradle/internal/concurrent/ExecutorFactory;Lorg/gradle/api/internal/file/temp/TemporaryFileProvider;)Lorg/gradle/process/internal/DefaultExecActionFactory;
```

**Comment by sksamuel** [comment]
I see that too.

**Comment by rezammalik** [comment]
I assume this is ok and is going to be only available on Gradle 7 and up moving forward?

**Comment by williamokano-dh** [comment]
Hi @sksamuel , when the snapshot version will be released? I tried the tags filter on `0.3.8` and it doesn't seem to be working, I'll assume that's because the version is not merged yet.

Tried both styles `./gradlew kotest --tags` and adding `kotest { tags.set("mytag") }`.

Thanks

**Comment by rezammalik** [comment]
The tag option definitely works on 0.3.8 and Gradle 7+. I have them defined like this in my test
```
object MyTag : Tag()

class MyClass : DescribeSpec({
tags(MyTag)
....
```
then I call it like this:

` ./gradlew clean kotest --tags MyTag`

**Comment by williamokano-dh** [comment]
Hi @rezammalik , yeah, it's working for sure. Somehow my dependencies weren't being correctly picked-up on my machine. I deleted all my dependencies and re-installed everything and it's indeed working.
